nmdc/reader.go

```diff
@@ -121,7 +121,7 @@ func (r *Reader) readMsgTo(ptr *Message) error {
 				Err: errors.New("message should not contain null characters"),
 			}
 		}
-		line = bytes.TrimSuffix(line, []byte("|"))
+		line = line[:len(line)-1] // trim delimiter
```

Comment:
Makes total sense, but let's check the delimiter (and the length of `line`) at least.

Right now it won't make a huge difference but may prevent bugs later if we change the code to omit the delimiter, for example.

Reply:
I've checked lineproto and it's impossible for a line to be returned without delimiter.
This eliminates multiple conditions and function calls (https://golang.org/src/bytes/bytes.go?s=16648:16688#L602) and helps catching eventual malfunctions in the upstream code. Staying on the safe side means preventing errors by acting on the error source and not on the dependent code.

Of course it's a very minor change and i don't want to lose time on it, i'm just reviewing the code and reporting everything, you can choose to accept it or o not if you think it's not safe enough.

Reply:
Yes, it's not possible currently because the reader prevents it. But to be on the safe side, I prefer to double-check it because these are two separate packages.

Anyway, I've added the check. Thanks for finding this slow spot!